### PR TITLE
BUG: Reduction UDF returning `np.array` hits error when used in grouped aggregation

### LINKED ISSUE #2768: BUG: Reduction UDF returning `np.array` hits error when used in grouped aggregation
```
import numpy as np
import pandas as pd

import ibis
import ibis.expr.datatypes as dt
import ibis.expr.operations as ops
import ibis.expr.types as ir
import ibis.udf.vectorized as udf

### Setup

df = pd.DataFrame({
    'key_col': pd.Series([1, 1, 2, 2,]),
    'int_col': pd.Series([100, 200, 300,]),
})

ibis_pandas = ibis.pandas.connect({'test': df})
t = ibis_pandas.table('test')


### The interesting part

@udf.reduction(
    input_type=[dt.int64],
    output_type=dt.Array(dt.int64)
)
def collect_udf(col):
    return np.array(col)

expr = t.groupby('key_col').aggregate(
    collected=collect_udf(t.int_col)
)

result = ibis_pandas.execute(expr)
```

```
...
~/ibis/ibis/backends/pandas/aggcontext.py in agg(self, grouped_data, function, *args, **kwargs)
    367             )
    368 
--> 369         return grouped_data.agg(wrap_for_agg(function, args, kwargs))
    370 
    371 

~/miniconda3/envs/ibis-dev/lib/python3.9/site-packages/pandas/core/groupby/generic.py in aggregate(self, func, engine, engine_kwargs, *args, **kwargs)
    261                 # TODO: KeyError is raised in _python_agg_general,
    262                 #  see test_groupby.test_basic
--> 263                 result = self._aggregate_named(func, *args, **kwargs)
    264 
    265             index = Index(sorted(result), name=self.grouper.names[0])

~/miniconda3/envs/ibis-dev/lib/python3.9/site-packages/pandas/core/groupby/generic.py in _aggregate_named(self, func, *args, **kwargs)
    482             if not initialized:
    483                 # We only do this validation on the first iteration
--> 484                 libreduction.check_result_array(output, 0)
    485                 initialized = True
    486             result[name] = output

pandas/_libs/reduction.pyx in pandas._libs.reduction.check_result_array()

pandas/_libs/reduction.pyx in pandas._libs.reduction.check_result_array()

ValueError: Must produce aggregated value
```

### PR DESCRIPTION
**Pull request by timothydijamco**
Closes #2768

### CODE REVIEW

File: ibis/backends/pandas/tests/test_udf.py [hunk 1]
```diff
@@ -408,6 +408,18 @@ def test_array_return_type_reduction_window(con, t, df, qs):
     tm.assert_series_equal(result, expected)
 
 
+def test_array_return_type_reduction_group_by(con, t, df, qs):
+    expr = t.groupby(t.key).aggregate(
+        quantiles_col=quantiles(t.b, quantiles=qs)
```
Comment (icexelloss) [inline ibis/backends/pandas/tests/test_udf.py:413]: Can quantiles return np array?

File: ibis/backends/pandas/aggcontext.py [hunk 2]
```diff
@@ -363,7 +363,13 @@ def agg(self, grouped_data, function, *args, **kwargs):
                 'Object {} is not callable or a string'.format(function)
             )
 
-        return grouped_data.agg(wrap_for_agg(function, args, kwargs))
+        if isinstance(grouped_data, pd.core.groupby.generic.SeriesGroupBy):
+            # Using `apply` as a workaround for the fact that
+            # `SeriesGroupBy.agg` does not allow np.arrays to be returned
+            # from UDFs.
+            return grouped_data.apply(wrap_for_agg(function, args, kwargs))
```
Comment (icexelloss) [inline ibis/backends/pandas/aggcontext.py:370]: In what case we want to exercise the "else" branch here?

File: ibis/backends/dask/tests/test_udf.py [hunk 3]
```diff
@@ -432,6 +432,19 @@ def test_array_return_type_reduction_window(con, t, df, qs):
     tm.assert_series_equal(result, expected)
 
 
+def test_array_return_type_reduction_group_by(con, t, df, qs):
+    expr = t.groupby(t.key).aggregate(
```
Comment (jreback) [inline ibis/backends/dask/tests/test_udf.py:436]: can you add the issue reference as a comment, also a 1-liner to explain the issue

File: ibis/backends/pandas/tests/test_udf.py [hunk 4]
```diff
@@ -408,6 +408,18 @@ def test_array_return_type_reduction_window(con, t, df, qs):
     tm.assert_series_equal(result, expected)
 
 
+def test_array_return_type_reduction_group_by(con, t, df, qs):
+    expr = t.groupby(t.key).aggregate(
```
Comment (jreback) [inline ibis/backends/pandas/tests/test_udf.py:412]: same as above

**Review by jreback — CHANGES_REQUESTED** [review]
can you add a release note as well

File: ibis/backends/pandas/aggcontext.py (hunk 2 above)
Comment (timothydijamco) [reply ibis/backends/pandas/aggcontext.py]: This is when exercised when doing an ungrouped aggregation, e.g. `my_arr_col.collect()`

File: ibis/backends/dask/tests/test_udf.py (hunk 3 above)
Comment (timothydijamco) [reply ibis/backends/dask/tests/test_udf.py:473]: done!

File: ibis/backends/pandas/tests/test_udf.py (hunk 1 above)
Comment (timothydijamco) [reply ibis/backends/pandas/tests/test_udf.py:420]: Ok, changed now.

**Review by jreback — APPROVED** [review]

**Pull request merged**